src/odemis/acq/acqmng.py

```diff
@@ -337,6 +339,26 @@ def foldStreams(streams, reuse=None):
         elif isinstance(s, ScannedRemoteTCStream):
             # Don't add extra FLIM streams
             continue
+        elif isinstance(s, SEMMDStream):
+            # Fold streams if roi, repetition and dwellTime are the same, but they all have different detectors
+            for smds in scan_semmds:
+                compatible = True
+                for smd in smds:
+                    # "compatible" means: same emitter/scanner/excitation
+                    if (smd._dwellTime.value == s._dwellTime.value and
+                            smd.repetition.value == s.repetition.value and
+                            smd.roi.value == s.roi.value and
+                            # as of now, each MDStream has only 2 streams
+                            smd.streams[1].detector.name != s.streams[1].detector.name
+                    ):
+                        compatible = True
```

Comment:
What this loop does is to look at a group of SEMMDStreams that are going to be folded, and see if it's possible to also add this (`smd`) Stream to the group. For this to be possible it should be compatible with all the other streams. 
So you shouldn't set `compatible` to True as soon as one stream is compatible. You should only do the `else` part (set to False). => Invert the `if` with a `not`.